tests/gaco.cpp

```diff
@@ -343,3 +345,46 @@ BOOST_AUTO_TEST_CASE(integer_test_2)
         std::all_of(x.begin(), x.end(), [](double el) { return (el == std::floor(el)); });
     }
 }
+
+BOOST_AUTO_TEST_CASE(bfe_usage_test)
+{
+    population pop{rosenbrock{10u}, 200u, 23u};
+    gaco uda{40u, 10u, 1.0, 25.0, 0.01, 5u, 7u, 1000u, 1000u, 0.0, 10u, 0.9, false, 23u};
+    uda.set_verbosity(1u);
+    uda.set_seed(23u);
+    uda.set_bfe(bfe{}); // This will use the default bfe.
+    pop = uda.evolve(pop);
+    
+    population pop_2{rosenbrock{10u}, 200u, 23u};
+    gaco uda_2{40u, 10u, 1.0, 25.0, 0.01, 5u, 7u, 1000u, 1000u, 0.0, 10u, 0.9, false, 23u};
+    uda_2.set_verbosity(1u);
+    uda_2.set_seed(23u);
+    pop_2 = uda_2.evolve(pop_2);
+    
+    BOOST_CHECK_EQUAL(pop.champion_f()[0], pop_2.champion_f()[0]);
+}
+
+BOOST_AUTO_TEST_CASE(bfe_performance_test)
+{
+    
+    auto start_time = std::chrono::high_resolution_clock::now();
+    population pop{lennard_jones{40u}, 200u, 23u};
+    gaco uda{10u, 10u, 1.0, 25.0, 0.01, 5u, 7u, 1000u, 1000u, 0.0, 10u, 0.9, false, 23u};
+    uda.set_seed(23u);
+    uda.set_bfe(bfe{});
+    pop = uda.evolve(pop);
+    auto end_time = std::chrono::high_resolution_clock::now();
+    std::cout << "gaco on Lennard Jones with bfe duration:" << std::endl;
+    std::cout << std::chrono::duration_cast<std::chrono::seconds>(end_time - start_time).count() << ":";
+    std::cout << std::chrono::duration_cast<std::chrono::microseconds>(end_time - start_time).count() << std::endl;
+    
+    auto start_time_2 = std::chrono::high_resolution_clock::now();
+    population pop_2{lennard_jones{40u}, 200u, 23u};
+    gaco uda_2{10u, 10u, 1.0, 25.0, 0.01, 5u, 7u, 1000u, 1000u, 0.0, 10u, 0.9, false, 23u};
+    uda_2.set_seed(23u);
+    pop_2 = uda_2.evolve(pop_2);
+    auto end_time_2 = std::chrono::high_resolution_clock::now();
+    std::cout << "gaco on Lennard Jones without bfe duration:" << std::endl;
+    std::cout << std::chrono::duration_cast<std::chrono::seconds>(end_time_2 - start_time_2).count() << ":";
+    std::cout << std::chrono::duration_cast<std::chrono::microseconds>(end_time_2 - start_time_2).count();
+}
```

Comment:
@bluescarni, @darioizzo 
lacking other examples on how to implement the test with `bfe`, I thought about adding a test in which the time with and w/o `bfe` is printed to screen in case a large population size is chosen, to actually show that the two times differ..  I could have done a check for the time (showing that with `bfe` it was indeed smaller), but then I do not think that this would be verified in all the machines..

I am not sure this is what you actually want .. I can change it as you wish or remove it in case it is not appropriate.

Reply:
PS. I just saw the first CI failure due to the lack of the 'b' when documenting `set_bfe`.. I will correct it in the next commit

Reply:
@Sceki unit tests are supposed to test the correctness of the code, so an appropriate test would be to check whether two evolutions with identical initial conditions (including random seeds), but one with ``bfe`` and the other without,  produce identical results.